src/build-defs.cmake

```diff
@@ -7,8 +7,8 @@ if(UNIX)
   set(CMAKE_C_STANDARD 11)
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -L${PG_LIBDIR}")
   set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -L${PG_LIBDIR}")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${PG_CFLAGS}")
-  set(CMAKE_CPP_FLAGS "${CMAKE_CPP_FLAGS} ${PG_CPPFLAGS}")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${PG_CFLAGS} ${PG_CPPFLAGS}")
```

Comment:
What are you building that needs C++ flags? TimescaleDB itself doesn't use them IIRC. Postgres only uses them for the JIT engine if you configure it with `--with-llvm`

This line in particular doesn't look correct, you can't generally put the C++ flags into the C ones.

Reply:
`pg_config` passes includes in the `CPPFLAGS` on my system it looks like this: `pg_config --cppflags
-I/opt/homebrew/Cellar/icu4c/72.1/include -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -I/opt/homebrew/opt/gettext/include -I/opt/homebrew/opt/openssl@1.1/include -I/opt/homebrew/opt/readline/include -I/opt/homebrew/Cellar/lz4/1.9.4/include` while OpenSSL and MacOSX13.sdk are correctly discovered in by CMake, it knows nothing about `icu4c` and `gettext` leading to #2777

Reply:
I realized that CPPFLAGS are C preprocessor flags, not C++ flags. CMake doesn't have support for them, so it's OK to just paste them with the rest of the C flags. So this line looks correct.